bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/DebugModeHandler.java

```diff
@@ -561,6 +563,31 @@ public void run() {
                 + ". If the debug connection timed out but the server did start successfully, you can still manually create a Remote Java Application debug configuration with the mentioned port and attach to the server.");
     }
 
+    /**
+     * Returns the liberty server module project associated with the input project.
+     * 
+     * @param project The project to process.
+     * 
+     * @return The liberty server module project associated with the input project.
+     * 
+     * @throws Exception
+     */
+    private Project getLibertyServerProject(Project project) throws Exception {
+        if (project.isParentOfServerModule()) {
```

Comment:
This seems like a good way to do it.  It's theoretically more restrictive than dev mode since you could have a child server module that's not defined in the aggregation, which dev mode would not treat as ambiguous but this check would.  But I think this is fine.. and much simpler to do.